odc/stats/plugins/gm.py

```diff
@@ -76,24 +78,31 @@ def native_transform(self, xx: xr.Dataset) -> xr.Dataset:
         if self._mask_band not in xx.data_vars:
             return xx
 
-        # Apply the contiguity flag
-        non_contiguent = xx.get("nbart_contiguity", 1) == 0
-
         # Erase Data Pixels for which mask == nodata
         mask = xx[self._mask_band]
         bad = enum_to_bool(mask, self._nodata_classes)
-        bad = bad | non_contiguent
+        # Apply the contiguity flag
+        if self._contiguity_band is not None:
+            non_contiguent = xx.get(self._contiguity_band, 1) == 0
+            bad = bad | non_contiguent
+
         if self.cloud_filters is not None:
             for cloud_class, c_filter in self.cloud_filters.items():
-                cloud_mask = enum_to_bool(mask, (cloud_class,))
+                if isinstance(cloud_class, tuple):
+                    cloud_mask = enum_to_bool(mask, cloud_class)
+                else:
+                    cloud_mask = enum_to_bool(mask, (cloud_class,))
```

Comment:
I usually prefer to "normalize" type of the variable using an if statement and then only have one "action line" afterwards when this is possible.

```suggestion
                if not isinstance(cloud_class, tuple):
                    cloud_class = (cloud_class,)
                cloud_mask = enum_to_bool(mask, cloud_class)
```

we are not doing something completely different when `cloud_class` is tuple vs string, we just treat a string and a tuple containing a single string as equivalent inputs.